### ISSUE TITLE
\include{doc} ignores EXAMPLE_PATH 

### ISSUE THREAD

**Issue by eseiler** [doxygen/doxygen#10449]
**Describe the bug**
Warning:
```text
example/include/header.hpp:3: warning: \include{doc} file 'example/include/fragments/example.md' not found
example/include/header.hpp:6: warning: Reached end of file while still inside a (nested) comment. Nesting level 1 (probable line reference: 1)
```

And the content of the file is not included.

`EXAMPLE_PATH` doesn't seem to be checked for the included file, instead the base path of the file that calls `\include` is used.

```
example/
├── Doxyfile
├── doc
│   └── fragments
│       └── example.md
└── include
    └── header.hpp
```

With `header.hpp`:
```cpp
/*!\brief A struct.
 * \details
 * \include{doc} fragments/example.md
 */
struct S{};
```

With `example.md`:
```md
This is some documentation to be included.
```

And an otherwise default `Doxyfile` with:
```diff
-INPUT                  =
+INPUT                  = include/

-EXAMPLE_PATH           =
+EXAMPLE_PATH           = doc/
```

It also does not work if I put the `md` file into `doc`, remove the `fragments` directory, and use the absolute root path as `EXAMPLE_PATH`, and then try to include `doc/...`.

**Expected behavior**
No warning, and content is included.

**To Reproduce**
[example.tar.gz](https://github.com/doxygen/doxygen/files/13472141/example.tar.gz)

**Version**
Persists on current master (c87ad3cbfbfe67a14e5078250178478719815dbb)

Bisect:
Works: e73e0f92124f47c77ea5b9d6d8c0dd49f6a7bbf1
Breaks: 7780d514980a97cfc2a5f21761ed7da338466350

**Comment by albert-github** [comment]
This looks like to have the same / similar problem as investigated in https://github.com/doxygen/doxygen/issues/7688#issuecomment-1822862189

**Comment by eseiler** [comment]
> This looks like to have the same / similar problem as investigated in https://github.com/doxygen/doxygen/issues/7688#issuecomment-1822862189

Indeed, didn't saw the comments when checking commits/PRs for bisecting :)

**Comment by albert-github** [comment]
Regarding the `EXAMPLE_PATH`: Please verify if the commit ([e8c5f55](https://github.com/doxygen/doxygen/commit/e8c5f5596fc73ee329dec03f0c8ebdd8247a10dd)) fixes the problem for you.

@ doxygen 
At the moment I get the impression that this is not yet the case though as I get the warnings:
```
.../example/include/header.hpp:3: warning: \include{doc} file 'D:/speeltuin/issue_10449/example/include/fragments/example.md' not found
.../example/include/header.hpp:6: warning: Reached end of file while still inside a (nested) comment. Nesting level 1 (probable line reference: 1)
```

**Comment by eseiler** [comment]
It doesn't quite fix the problem.

I get the same warning you described:

```
/dev/shm/example/include/header.hpp:3: warning: \include{doc} file '/dev/shm/example/include/fragments/example.md' not found
/dev/shm/example/include/header.hpp:6: warning: Reached end of file while still inside a (nested) comment. Nesting level 1 (probable line reference: 1)
```

When setting `RECURSIVE = YES` in the `Doxyfile`, I get 
```
/dev/shm/example/include/header.hpp:3: warning: \include{doc} file 'fragments/example.md' could not be read
```

Replacing `\include{doc} fragments/example.md` with `\include{doc} doc/fragments/example.md` makes it work, independent of `RECURSIVE = YES/NO`.

---

However, the workaround of adding `doc/` does not work if doxygen is being run inside a subdirectory.
This is what actually happens in the repository from which I derived the minimal example.

The warning is:
```
/dev/shm/example/include/header.hpp:3: warning: \include{doc} file 'doc/fragments/example.md' could not be read
```

Here is another test case:

```
example
├── doc
│   └── fragments
│       └── example.md
├── include
│   └── header.hpp
└── test
    └── Doxyfile
```

example.md (same as in original test case)
```
This is some documentation to be included.
```

header.hpp (`doc/` added)
```cpp
/*!\brief A struct.
 * \details
 * \include{doc} doc/fragments/example.md
 */
struct S{};
```

Doxyfile (changed, diff to default)
```diff
-INPUT                  =
+INPUT                  = ../include

-RECURSIVE              = NO
+RECURSIVE              = YES

-EXAMPLE_PATH           =
+EXAMPLE_PATH           = ../doc
```

`1.9.8` accepts both `../doc` and `..` as `EXAMPLE_PATH`.

[example2.tar.gz](https://github.com/doxygen/doxygen/files/13500260/example2.tar.gz)

**Comment by albert-github** [comment]
@doxygen I still get a warning like
```
.../ex_1/include/header.hpp:3: warning: \include{doc} file 'D:/speeltuin/issue_10449/ex_1/include/fragments/example.md' not found
.../include/header.hpp:6: warning: Reached end of file while still inside a (nested) comment. Nesting level 1 (probable line reference: 1)
```

Example: [example.tar.gz](https://github.com/doxygen/doxygen/files/13536533/example.tar.gz)

Note: the second warning is also mentioned as note in the proposed pull request #10457

**Comment by eseiler** [comment]
> @doxygen I still get warnings like:
> 
> ```
> .../include/header.hpp:3: warning: \include{doc} file '.../include/fragments/example.md' not found
> .../include/header.hpp:6: warning: Reached end of file while still inside a (nested) comment. Nesting level 1 (probable line reference: 1)
> ```
> 
> Example: [example.tar.gz](https://github.com/doxygen/doxygen/files/13536533/example.tar.gz)
> 
> Note: the second warning is also mentioned as note in the proposed pull request #10457

It works with `RECURSIVE = YES`. 

Since we use `RECURSIVE = YES`, our project builds again.

**Comment by albert-github** [comment]
Interesting to see here also the behavior of the `\include` command.
In case of `RECURSIVE=NO`  the `\include fragments/example.md` works as as expected and does not throw a warning.

**Comment by doxygen** [comment]
@albert-github It does indeed not make much sense to have different results for `\include` and `\include{doc}` for the same file. So I've now used the same routine for resolving the name for `\include{doc}` as was there for `\include`.

**Comment by albert-github** [comment]
Using the same procedure for `\include{doc}` and `include` definitely makes sense.
Have to do some testing to seen whether or not everything works now so as expected.

**Comment by doxygen** [comment]
@albert-github One more commit to relax the absolute path requirement.

**Comment by albert-github** [comment]
I get a warning during compilation:
```
...\src\util.cpp(3293): warning C4456: declaration of 'fi' hides previous local declaration
...\src\util.cpp(3281): note: see declaration of 'fi'
```

**Comment by doxygen** [comment]
@albert-github Somehow my compiler doesn't produce this warning. Should be fixed now.

**Comment by albert-github** [comment]
Indeed fixed, but the code like:
```
  bool found=false;
  if (!found)
```
will probably be flagged by sonarcloud or coverity as "code smell" as the `if` is a bit useless, I think leaving the block but removing the `if` would be better.
